Dockerfile

```diff
@@ -0,0 +1,50 @@
+ARG RUBY_VERSION=3.2.0
+ARG DISTRO_NAME=bullseye
+
+FROM --platform=${BUILDPLATFORM:-linux/amd64} ruby:$RUBY_VERSION-slim-$DISTRO_NAME AS base
+
+WORKDIR /applelink
+
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y \
+    curl \
+    libjemalloc2 \
+    gnupg2 \
+    less \
+    git \
+    libnss3-tools && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+FROM --platform=${BUILDPLATFORM:-linux/amd64} base AS builder
+
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y \
+    build-essential \
+    pkg-config \
+    libvips-dev && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY Gemfile Gemfile.lock .ruby-version ./
+
+RUN gem install bundler -v 2.4.2 && \
+    bundle config set --local without 'development test' && \
+    bundle install --jobs=4 --retry=3
+
+COPY . .
+
+FROM --platform=${TARGETPLATFORM:-linux/amd64} base
+
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y \
+    libvips42 && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
Why run these `apt-get` installs and cleanups in 3 different sections? 

Why not just install everything together in one place?

Reply:
Thanks for flagging this! I've now consolidated all system package installs into a single `apt-get` install block in a shared runtime base (`runtime`). This removes the visual repetition across stages while still using a multi-stage build to keep build-time deps isolated. Let me know if you'd prefer going back to a more aggressively stripped runtime image, happy to do that too!